packages/commonwealth/server/scripts/setupAppRoutes.ts

```diff
@@ -133,92 +133,96 @@ const setupAppRoutes = (
     renderWithMetaTags(res, title, description, author, image);
   });
 
-  const renderProposal = async (
+  const renderThread = async (
     scope: string,
-    proposalType: string,
-    proposalId: string,
+    threadId: string,
     res,
     chain?: ChainInstance
   ) => {
-    // Retrieve title, description, and author from the database
-    let title, description, author, image;
     chain = chain || (await models.Chain.findOne({ where: { id: scope } }));
 
-    if (proposalType === 'discussion' && proposalId !== null) {
-      // Retrieve discussions
-      const proposal = await models.Thread.findOne({
-        where: { id: proposalId },
-        include: [
-          {
-            model: models.Chain,
-          },
-          {
-            model: models.Address,
-            as: 'Address',
-          },
-        ],
-      });
+    // Retrieve discussions
+    const thread = await models.Thread.findOne({
+      where: { id: threadId },
+      include: [
+        {
+          model: models.Chain,
+        },
+        {
+          model: models.Address,
+          as: 'Address',
+        },
+      ],
+    });
 
-      title = proposal ? decodeTitle(proposal.title) : '';
+    const title = thread ? decodeTitle(thread.title) : '';
+    const description = thread ? thread.plaintext : '';
+    const image = chain
+      ? `https://commonwealth.im${chain.icon_url}`
+      : DEFAULT_COMMONWEALTH_LOGO;
 
-      description = proposal ? proposal.plaintext : '';
-      image = chain
-        ? `https://commonwealth.im${chain.icon_url}`
-        : DEFAULT_COMMONWEALTH_LOGO;
-      try {
-        const profile = await models.Profile.findOne({
-          where: { id: proposal.Address.id },
-        });
-        author = profile.profile_name;
-      } catch (e) {
-        author = '';
-      }
-    } else {
-      title = chain ? chain.name : 'Commonwealth';
-      description = '';
-      image = chain
-        ? `https://commonwealth.im${chain.icon_url}`
-        : DEFAULT_COMMONWEALTH_LOGO;
+    let author;
+    try {
+      const profile = await models.Profile.findOne({
+        where: { id: thread.Address.id },
+      });
+      author = profile.profile_name;
+    } catch (e) {
       author = '';
     }
+
+    renderWithMetaTags(res, title, description, author, image);
+  };
+
+  const renderProposal = async (
+    scope: string,
+    res,
+    chain?: ChainInstance
+  ) => {
+    // Retrieve title, description, and author from the database
+    chain = chain || (await models.Chain.findOne({ where: { id: scope } }));
```

Comment:
Can we batch this repeated code into a helper function?

Reply:
Unsure what you mean by batch. Only one call is made per request. Do you mean to refactor this line of code out into its own helper function for code clarity reasons?

Reply:
Yes -- we do the same function in both renderProposal and renderThread, so we should move it into a helper function.

Reply:
Unsure what you mean by batch. This line is only called once. Is the suggestion to move this line into its own function for code clarity?

Reply:
I had removed this part and joined it in with the thread query we were making.